pkg/api/graph/interfaces.go

```diff
@@ -51,6 +53,57 @@ func (m Markers) BySeverity(severity Severity) []Marker {
 	return ret
 }
 
+// FilterCurrentNamespaceMarkers returns all the markers that are not associated with missing nodes
+// from other namespaces (other than the currentNamespace).
+func (m Markers) FilterCurrentNamespaceMarkers(currentNamespace string) Markers {
+	// Create a subgraph with all involved nodes.
+	nodes := map[int]graph.Node{}
+	for i := range m {
+		nodes[m[i].Node.ID()] = m[i].Node
```

Comment:
Can't you do the synthetic test right here?  Type assertion for `ExistenceChecker`, then check `Found`, then type assertion for  `objectifier`, then call `.Object` do the meta check?  I might make a helper for it, but I don't see why I need to construct the graph.